nodejs/week3/meal-app/src/backend/api/meals.js

```diff
@@ -0,0 +1,169 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const pool = require("./../database");
+
+
+router.get("/", (req, res) => {
+    const maxPrice = req.query.maxPrice;
+    const availableReservations = req.query.availableReservations;
+    const title = req.query.title;
+    const createdAfter = req.query.createdAfter;
+    const limit = req.query.limit;
+
+    if (Object.entries(req.query).length === 0) {
+        pool.query("select * from meal", function(error, results, fields) {
+            if (error) {
+                console.log(error);
+            }
+            res.send(results);
+            // http://localhost:5000/api/meals
+        })
+    } else if (maxPrice) {
+        pool.query(`select * from meal where price < ${parseFloat(maxPrice)}`, function(error, results, fields) {
+            if (error) {
+                console.log(error);
+            }
+            res.send(results);
+            // http://localhost:5000/api/meals?maxPrice=50
+        })
+    } else if (availableReservations && JSON.parse(availableReservations) === true) {
+        pool.query(`select meal.*, reservation.number_of_guests
+         from meal inner join reservation on reservation.meal_id = meal.id where meal.max_reservations > reservation.number_of_guests;`, function(error, results, fields) {
+            if (error) {
+                console.log(error);
+            }
+            res.send(results);
+            // http://localhost:5000/api/meals?availableReservations=true
+        })
+    } else if (title) {
+        pool.query(`select * 
+        from meal
+        where title like '%${title}%'`, function(error, results, fields) {
```

Comment:
There is a similar SQL injection issue here. Could be solved like this:

```suggestion
        where title like ?'`, `%${title}%` function(error, results, fields) {
```